### PR TITLE
Fix for NaN's when training DASM with ambiguous sequences

### PR DESCRIPTION
**Pull request by willdumm**
The fix is one line, applying a mask that should have been applied from the very beginning, but previously didn't matter much since the data didn't contain ambiguities.

Also adds parallelized neutral model application, moved to the cpu. This makes running some notebooks much faster, and also speeds up setup for training Snakemake runs.

### CODE REVIEW

File: netam/dasm.py [hunk 1]
```diff
@@ -195,11 +196,10 @@ def loss_of_batch(self, batch):
         # logit space, so we are set up for using the cross entropy loss.
         # However we have to mask out the sites that are not substituted, i.e.
         # the sites for which aa_subs_indicator is 0.
-        subs_mask = aa_subs_indicator == 1
+        subs_mask = (aa_subs_indicator == 1) & mask
```
Comment (willdumm) [inline netam/dasm.py:199]: This is it, the whole Nan/inf fix! Could you have a look around here and make sure this looks reasonable to you, too?

Comment (matsen) [reply netam/dasm.py:199]: Nice sleuthing! It seems good to me... am I missing something?

**Review by matsen — APPROVED** [review]
Awesome!

File: netam/dasm.py (hunk 1 above)
Comment (willdumm) [reply netam/dasm.py:199]: I don't think so, just wanted extra eyes on it! Thanks

**Pull request merged**